src/client.js

```diff
@@ -251,12 +251,13 @@ assign(Client.prototype, {
 
     return Object.assign(poolConfig, {
       create: () => {
-        return this.acquireRawConnection().tap((connection) => {
+        return this.acquireRawConnection().then((connection) => {
           connection.__knexUid = uniqueId('__knexUid');
 
           if (poolConfig.afterCreate) {
             return Bluebird.promisify(poolConfig.afterCreate)(connection);
```

Comment:
Isn't this part going to mess up with return functionality? Probably return call should be replace with await here.